src/config.rs

```diff
@@ -37,6 +73,26 @@ impl Display for OverrideReason {
     }
 }
 
+#[derive(Default)]
+struct OverrideCfg<'a> {
+    toolchain: Option<Toolchain<'a>>,
+    components: Vec<String>,
+    targets: Vec<String>,
+}
+
+impl<'a> OverrideCfg<'a> {
+    fn from_file(cfg: &'a Cfg, file: OverrideFile) -> Result<Self> {
+        Ok(Self {
```

Comment:
Does a result (None,None,None) make sense? what will happen when someone has such a config file - will it produce an error? I see the merit in permitting configs where at least one of the fields is not none, but with all none - do we want to treat that as an absent file, or as an error? @kinnison

Reply:
I went for treating this case as an empty file in the legacy format, which was not yielding errors.

Reply:
A file consisting of `[toolchain]` only (i.e. the `(None, None, None)` case) is an error IMO.  You must be specifying *something* to warrant a `rust-toolchain` file - similarly, an empty such file is worthless and thus an error as I mentioned in a previous comment.

Reply:
(While previously an empty file was not an error, I think from now on it *should* be.  and I'm prepared to accept that as a breaking change if people view it as such.)

Reply:
With the latest version `OverrideFile` is guaranteed to have at least one `Some`.